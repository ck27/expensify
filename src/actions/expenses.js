import {v4 as uuidV4} from "uuid";

export const addExpense = ( { description = "", note = "", amount= 0, createdAt = 0 } = {}) => ({
    type: "ADD_EXPENSE",
    expense: {
        id: uuidV4(),
        description,
        note,
        amount,
        createdAt
    }
});

export const deleteExpense = ( { id } = {} ) => ( {
    type:"DELETE_EXPENSE",
    id
});

export const editExpense = ( expense, updates ) => ({
    type: "EDIT_EXPENSE",
    expense,
    updates
});