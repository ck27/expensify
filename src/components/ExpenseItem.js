import React from "react";
import { connect } from "react-redux";
import { Link } from "react-router-dom";

const ExpenseItem = ({ id, description, amount, createdAt, dispatch}) => (
    <div>
        <Link to={`/edit/${id}`}>
            <h3>{description}</h3>
        </Link>
        <div>
            <p>{amount} Paid at {createdAt}</p>
        </div>
    </div>
);

export default ExpenseItem;