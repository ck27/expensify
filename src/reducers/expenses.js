const expensesReducerDefaultState = [];

export const expensesReducer = (state = expensesReducerDefaultState, action) => {
    switch(action.type){
        case "ADD_EXPENSE":
            return [
                ...state,
                action.expense
            ];
        case "DELETE_EXPENSE":
            return state.filter( ( {id} ) => id !== action.expenseId );

        case "EDIT_EXPENSE":
            return state.map( (exp) => {
                if(exp.id == action.expense.id) {
                    return {
                        ...exp,
                        ...action.expense
                    };
                } else {
                    return exp;
                }
            });
        default:
            return state;
    }

};
