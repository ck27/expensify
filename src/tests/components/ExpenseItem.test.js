import React from "react";
import {shallow} from "enzyme";
import ExpenseItem from "../../components/ExpenseItem";
import expenses from "../fixtures/expenses";

test("it should render an expense item", () => {
    const content = shallow( <ExpenseItem {...expenses[0]} />);
    expect(content).toMatchSnapshot();

})