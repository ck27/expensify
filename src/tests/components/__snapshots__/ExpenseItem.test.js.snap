// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it should render an expense item 1`] = `
<div>
  <Link
    to="/edit/1"
  >
    <h3>
      Rent
    </h3>
  </Link>
  <div>
    <p>
      21000
       Paid at 
      -345600000
    </p>
  </div>
</div>
`;
